underfs/swift/src/main/java/alluxio/underfs/swift/SwiftInputStream.java

```diff
@@ -83,6 +93,9 @@ public int read(byte[] b, int offset, int length) throws IOException {
     int read = mStream.read(b, offset, length);
     if (read != -1) {
       mPos += read;
+      if (mPos % getBlockSize() == 0) {
+        closeStream();
+      }
```

Comment:
verified: the swift library truncates the stream to fileLength (8 in this case) => no need for explicit min(fileLength, endPos).